DuckDuckGo/TabViewController.swift

```diff
@@ -859,12 +859,13 @@ extension TabViewController: WKNavigationDelegate {
         
         if !failingUrls.contains(url.host ?? ""),
             navigationAction.isTargetingMainFrame(),
-            let upgradeUrl = httpsUpgrade.upgrade(url: url) {
+            let upgradeUrl = httpsUpgrade.upgrade(url: url),
+            lastUpgradedURL != upgradeUrl {
```

Comment:
If upgraded url matches the one from the last upgrade attempt, it means that something went wrong and we should not upgrade this time.